dev/AppNotifications/AppNotificationContentBuilder/AppNotificationContentBuilder.idl

```diff
@@ -0,0 +1,133 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace Microsoft.Windows.AppNotifications
+{
+    // Use for retrieving values store in the arguments string received from an AppNotification activation.
+    [contract(AppNotificationsContract, 2)]
+    runtimeclass ArgumentsDecoder
+    {
+        ArgumentsDecoder(String arguments);
+
+        String GetStringArgument(String key);
+        Int32 GetInt32Argument(String key);
+        Double GetDoubleArgument(String key);
+        Single GetSingleArgument(String key);
+        Boolean GetBooleanArgument(String key);
+    }
+
+    [contract(AppNotificationsContract, 2)]
+    runtimeclass ButtonConfig
+    {
+        ButtonConfig(String content);
+
+        // Gets the text to display on the button.
+        String Content{ get; };
+
+        // Gets app-defined string of arguments that the app can later retrieve once it is activated when the user clicks the button.
+        String Arguments { get; };
+
+        // Gets or sets an identifier used in telemetry to identify your category of action.This should be something like "Delete",
+        // "Reply", or "Archive".In the upcoming toast telemetry dashboard in Dev Center, you will be able to view how frequently
+        // your actions are being clicked.
```

Comment:
If this is just for telemetry, then as a dev why would I care about setting this property? Can you clarify the actual intention of this property?

Reply:
That's not for telemetry. The hint is used by Shell to organize the various elements inside the toast when it's displayed.